### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by georgiStoychev17**
Homework2 
Georgi Stoychev 7MI0600157

### CODE REVIEW

File: SI/Sem.02/Tasks2/Task1.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using namespace std;
+
+ int main() {
+
+	int yearNumber;
+	cout << "Year: ";
+	cin >> yearNumber;
+
+
+	if ((yearNumber % 4 == 0) && (yearNumber % 400 == 0) || (yearNumber % 100 != 0)) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task1.cpp:11]: Wrong if condition try 1901

File: SI/Sem.02/Tasks2/Task10.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	double number1, number2; // op = operator
+	char op;
+	
+	cout << "Enter two numbers: ";
+	cin >> number1 >> number2;
+	
+	cout << "Enter an operator: ";
+	cin >> op;
+	if (number2 != 0) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task10.cpp:14]: The if condition number2 != 0 should be only in the '/' case

File: SI/Sem.02/Tasks2/Task11.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+using namespace std;
+
+int main() {
+
+	int numberOne, numberTwo, numberThree;
+cin >> numberOne >> numberTwo >> numberThree;
+
+if (numberOne > numberTwo && numberTwo > numberThree) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task11.cpp:9]: Think of an easier way to solve this task
Hint: swap the numbers between them

File: SI/Sem.02/Tasks2/Task2.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+	char symbol;
+	cout << "Enter a symbol: ";
+	cin >> symbol;
+
+
+	if ((symbol >= 65 && symbol <= 90) || (symbol >= 97 && symbol <= 122)) {
+		switch (symbol) {
+
+		case 'a': cout << "That is a part of the latin alphabet and also a vowel."; break;
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task2.cpp:14]: You can reuse the cout logic like this
case 1:
case 2:
case 3: cout<<"....."
In each of the cases 1, 2, 3 we will execute the cout

File: SI/Sem.02/Tasks2/Task2.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		case 'E': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'o': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'O': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'u': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'U': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'i': cout << "That is a part of the latin alphabet and also a vowel."; break;
+		case 'I': cout << "That is a part of the latin alphabet and also a vowel."; break;
+
+		default:
+			cout << "That is a part of the latin alphabet. ";
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task2.cpp:26]: "This is consonant"

File: SI/Sem.02/Tasks2/Task4.cpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+int main() {
+
+	
+	int examPercentage;
+	cout << "Result: ";
+	cin >> examPercentage;
+	
+	if (examPercentage < 40) {
+
+		cout << "Grade: " << 2;
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task4.cpp:13]: You can put the 2 inside the string
"Grade: 2"

File: SI/Sem.02/Tasks2/Task9.cpp [hunk 7]
```diff
@@ -0,0 +1,40 @@
+#include <iostream>
+using namespace std;
+
+int main() {
+
+	int numberOne, numberTwo, numberThree;
+	cout << "Enther three whole numbers: ";
+	cin >> numberOne >> numberTwo >> numberThree;
+
+	if (numberOne > numberTwo && numberTwo > numberThree) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task9.cpp:10]: Think of an easier way to solve this task
Hint: swap the numbers between them

File: SI/Sem.02/Tasks2/Task12.cpp [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+		month = month++;
+		cout << "Next date: " << day << " " << month++ << endl;
+	}
+	else if (((month == 4) || (month == 6) || (month == 9) || (month == 11)) && (day == 30)) {
+
+		day = 1;
+		month = month++;
+		cout << "Next date: " << day << " " << month++ << endl;
+	}
+	else if ((1 <= day < 31) && (1 <= month <= 12)) {
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task12.cpp:33]: This solution does not cover all the cases try entering: 31 4 or 29 2

File: SI/Sem.02/Tasks2/Task3.cpp [hunk 9]
```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+using namespace std;
+
+int main() {
+
+	int firstNumber, secondNumber, condition;
```
Comment (GeorgiTerziev02) [inline SI/Sem.02/Tasks2/Task3.cpp:6]: I am not sure what is all this logic for
The actual solution is much simple
Check this solution: https://github.com/Justsvetoslavov/Introduction_to_programming/pull/22/files#diff-fb15057d198d4557d8abfe0f9118fb3ba1bde4dcdcc0b5d457cc8bfcc1ae1816

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]
Check the comments and refactor.

**Comment by Justsvetoslavov** [comment]
Review the new written PR rules and open new PRs if you would like (also check which ones are already merged ,as they will be duplicates).
[You can view them here](https://github.com/Justsvetoslavov/Introduction_to_programming)

**Pull request closed without merging**